build.cmd

```diff
@@ -13,6 +13,17 @@ if not defined MSBUILDCUSTOMPATH (
     set MSBUILDCUSTOMPATH=MSBuild.exe
 )
 
+set NUGETEXEPATH="%~dp0packages\NuGet.exe"
+
+if not exist "%NUGETEXEPATH%" (
+    :: This will need to be fixed for non-windows
+    echo ** Downloading NuGet.exe from https://nuget.org/NuGet.exe...
```

Comment:
Looks like you need to mkdir %~dp0packages

From the failed build:

** Downloading NuGet.exe from https://nuget.org/NuGet.exe...
PS> Invoke-WebRequest -OutFile "d:\j\workspace\microsoft_msbuild_prtest\packages\NuGet.exe" "https://nuget.org/NuGet.exe"
Invoke-WebRequest : Could not find a part of the path 
'D:\j\workspace\microsoft_msbuild_prtest\packages\NuGet.exe'.
At line:1 char:1
+ Invoke-WebRequest -OutFile 
d:\j\workspace\microsoft_msbuild_prtest\packages\NuGe ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~
    + CategoryInfo          : NotSpecified: (:) [Invoke-WebRequest], Directory 
   NotFoundException
    + FullyQualifiedErrorId : System.IO.DirectoryNotFoundException,Microsoft.P 
   owerShell.Commands.InvokeWebRequestCommand